### ISSUE TITLE
Harvester crashes:  ERROR    Exception:  <class 'asyncio.exceptions.TimeoutError'>, closing connection

### ISSUE THREAD

**Issue by Furious-George** [Chia-Network/chia-blockchain#13700]
I start my harvesters via systemctl:

`ExecStart=/bin/sh -c chia start harvester (code=exited, status=0/SUCCESS)`

One of my harvesters crashes within a few days of starting it.  This started when I moved my drives to a new server, with chia installed from an apt package rather than from source.

> 2022-10-12T16:15:13.484 harvester chia.plotting.util      : WARNING  Directory: /chia06 does not exist.
> 2022-10-12T16:15:13.487 harvester chia.plotting.util      : WARNING  Directory: /chia08 does not exist.
> 2022-10-12T16:15:47.452 harvester chia.harvester.harvester: WARNING  Looking up qualities on /chia18/plot
> 2022-10-12T16:16:29.799 harvester chia.harvester.harvester: WARNING  Looking up qualities on /chia18/plot
> 2022-10-12T16:16:40.678 harvester chia.harvester.harvester: WARNING  Looking up qualities on /chia18/plot
> 2022-10-12T16:17:19.053 harvester chia.harvester.harvester: WARNING  Looking up qualities on /chia18/plot
> 2022-10-12T16:18:49.876 harvester chia.harvester.harvester: WARNING  Looking up qualities on /chia18/plot
> 2022-10-12T16:24:59.855 harvester harvester_server        : ERROR    Exception:  <class asyncio.exceptio>
>   File "chia/server/server.py", line 608, in api_call
>   File "asyncio/tasks.py", line 501, in wait_for
> asyncio.exceptions.TimeoutError
> 
> 2022-10-12T16:24:59.859 harvester harvester_server        : WARNING  Banning 10.5.0.21 for 10 seconds
> 2022-10-12T16:25:35.606 daemon chia.daemon.server         : ERROR    Ping error
> Traceback (most recent call last):
>   File "chia/daemon/server.py", line 277, in ping_task
>   File "aiohttp/web_ws.py", line 295, in ping
>   File "aiohttp/http_websocket.py", line 676, in ping
>   File "aiohttp/http_websocket.py", line 653, in _send_frame
>   File "aiohttp/http_websocket.py", line 663, in _write
> ConnectionResetError: Cannot write to closing transport
> 2022-10-12T16:25:35.607 daemon chia.daemon.server         : ERROR    Ping failed, connection closed.

Note the line about banning 10.5.0.21.  That's odd, as it's my farmer's IP.

The harvester process becomes defunct, and it's impossible to restart the harvester without restarting the server:

> ps ax|grep chia
>   23499 ?        Sl     0:05 chia_daemon  
>   24408 ?        Sl    23:45 chia_harvester
>   1922361 pts/0    S+     0:00 grep chia
>
> chia stop all -d
> Error, timeout.
> Traceback (most recent call last):
>   File "chia/cmds/chia.py", line 153, in <module>
>   File "chia/cmds/chia.py", line 149, in main
>   File "click/core.py", line 829, in __call__
>   File "click/core.py", line 782, in main
>   File "click/core.py", line 1259, in invoke
>   File "click/core.py", line 1066, in invoke
>   File "click/core.py", line 610, in invoke
>   File "click/decorators.py", line 21, in new_func
>   File "chia/cmds/stop.py", line 55, in stop_cmd
>   File "asyncio/runners.py", line 44, in run
>   File "asyncio/base_events.py", line 616, in run_until_complete
>   File "chia/cmds/stop.py", line 23, in async_stop
> AttributeError: 'NoneType' object has no attribute 'get'
> [1922411] Failed to execute script 'chia' due to unhandled exception!
>
> ps ax|grep chia
>   23499 ?        Sl     0:05 chia_daemon
>   24408 ?        Zl    23:45 [chia_harvester] <defunct>
>   1922605 pts/0    S+     0:00 grep chia
>
>  kill -9 23499
>
>  kill -9 24408
>
>  ps ax|grep chia
>   24408 ?        Zl    23:45 [chia_harvester] <defunct>
>   1922707 pts/0    S+     0:00 grep chia

**Comment by wallentx** [comment]
Are you able to provide us with DEBUG logs? You may find a bit more stability if you clean up your plot directories to ensure that all listed paths are actually available. You may also want to check the temperatures of your drives. It could be that when you relocated your drives, one of them isn't getting enough ventilation, and is hanging up your process.

**Comment by Furious-George** [comment]
Hi again.

I'm not able to see drives overheating and though a few drives have an error or two in smartctl, they do not seem to be failing. 

Here is the output of the latest crash:

`2023-01-20T08:50:51.958 daemon chia.daemon.server         : DEBUG    About to ping: chia_harvester
2023-01-20T08:50:58.639 harvester harvester_server        : DEBUG    <- new_signage_point_harvester from peer 201d3f186882a7c571ba5784d07580358e79ac08aafb1ddb4bf045333e916d0a 10.5.0.21
2023-01-20T08:50:58.640 harvester chia.harvester.harvester: DEBUG    new_signage_point_harvester lookup: challenge_hash: 572f391a71e344d049d959361a56fdcbd0540fba3670defe06c757706040799d, sp_hash: 71f186c6b0af7b70df4d72320726a94973abd88c316e75976f1a00b7fb765665, signage_point_index: 25
2023-01-20T08:50:58.640 harvester chia.harvester.harvester: DEBUG    new_signage_point_harvester lock acquired
2023-01-20T08:50:58.835 harvester chia.harvester.harvester: DEBUG    new_signage_point_harvester 10 plots passed the plot filter
2023-01-20T08:50:59.414 harvester chia.harvester.harvester: INFO     10 plots were eligible for farming 572f391a71... Found 0 proofs. Time: 0.77449 s. Total 5034 plots
2023-01-20T08:50:59.415 harvester harvester_server        : DEBUG    -> farming_info to peer 10.5.0.21 201d3f186882a7c571ba5784d07580358e79ac08aafb1ddb4bf045333e916d0a
2023-01-20T08:50:59.415 harvester harvester_server        : DEBUG    Time taken to process new_signage_point_harvester from 201d3f186882a7c571ba5784d07580358e79ac08aafb1ddb4bf045333e916d0a is 0.7763848304748535 seconds
2023-01-20T08:50:59.416 daemon chia.daemon.server         : DEBUG    Received message: WSMessage(type=<WSMsgType.TEXT: 1>, data='{"ack": false, "command": "farming_info", "data": {"challenge_hash": "572f391a71e344d049d959361a56fdcbd0540fba3670defe06c757706040799d", "eligible_plots": 10, "found_proofs": 0, "success": true, "time": 0.7744929790496826, "total_plots": 5034}, "destination": "metrics", "origin": "chia_harvester", "request_id": "23ea6bc1826eb2bcb19f665fd3d062040fd02775310b83c3160c2be1b7522393"}', extra='')
2023-01-20T08:51:06.850 harvester harvester_server        : ERROR    Exception:  <class 'asyncio.exceptions.TimeoutError'>, closing connection {'host': '10.5.0.21', 'port': 8447}. Traceback (most recent call last):
  File "chia/server/server.py", line 634, in api_call
  File "asyncio/tasks.py", line 501, in wait_for
asyncio.exceptions.TimeoutError

2023-01-20T08:51:06.850 harvester harvester_server        : DEBUG    Inbound_handler task cancelled
2023-01-20T08:51:06.852 harvester harvester_server        : WARNING  Banning 10.5.0.21 for 10 seconds
2023-01-20T08:51:06.852 harvester harvester_server        : INFO     Connection closed: 10.5.0.21, node id: 201d3f186882a7c571ba5784d07580358e79ac08aafb1ddb4bf045333e916d0a
2023-01-20T08:51:06.852 harvester chia.harvester.harvester: INFO     peer disconnected {'host': '10.5.0.21', 'port': 8447}
2023-01-20T08:51:21.982 daemon chia.daemon.server         : DEBUG    About to ping: chia_harvester
2023-01-20T08:51:51.989 daemon chia.daemon.server         : DEBUG    About to ping: chia_harvester
2023-01-20T08:51:51.990 daemon chia.daemon.server         : ERROR    Ping error
Traceback (most recent call last):
  File "chia/daemon/server.py", line 283, in ping_task
  File "aiohttp/web_ws.py", line 295, in ping
  File "aiohttp/http_websocket.py", line 676, in ping
  File "aiohttp/http_websocket.py", line 653, in _send_frame
  File "aiohttp/http_websocket.py", line 663, in _write
ConnectionResetError: Cannot write to closing transport
2023-01-20T08:51:51.991 daemon chia.daemon.server         : ERROR    Ping failed, connection closed.`

Note that this harvester has the newer drives, whereas another harvester with a mishmash of ancient sata disks does not have this problem.

In this case it lasted less than 12 hours before it crashed again.

Looking at /var/log/messages, which overlaps the poiint at which the harvester crashed, I see no problems with the disks:

`# tail -fn 10  /var/log/messages
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129615]  entry_SYSCALL_64_after_hwframe+0x61/0xcb
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129619] RIP: 0033:0x7f78bafad387
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129621] RSP: 002b:00007f78b2126b68 EFLAGS: 00000293 ORIG_RAX: 00000000000000d9
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129623] RAX: ffffffffffffffda RBX: 00007f78a403cf20 RCX: 00007f78bafad387
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129625] RDX: 0000000000008000 RSI: 00007f78a403cf50 RDI: 000000000000000d
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129626] RBP: 00007f78a403cf50 R08: 0000000000000030 R09: 0000000000000001
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129627] R10: 0000000000000008 R11: 0000000000000293 R12: ffffffffffffff80
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129628] R13: 00007f78a403cf24 R14: 0000000000000000 R15: 0000000000000000
Jan 20 08:48:35 brickhouse-0 kernel: [59331.129630]  </TASK>
Jan 20 16:44:24 brickhouse-0 kernel: [87881.183184] perf: interrupt took too long (2540 > 2500), lowering kernel.perf_event_max_sample_rate to 78500`

As I mentioned before, I don't see evidence of a hardware failure with respect to the disks.  I reinstalled the OS on both harvesters when I went from source compiled to aptitude based installation, which is when this problem started.  

It is possible I screwed something up on this particular server with repspect to installation via aptitude, as I do remember wiping the ~/.chia dir and starting over via chia init.   Unfortunately I don't recall exactly what went wrong that caused me to do that. 

Thanks.
